### PR TITLE
Install `unzip` and `gh` command functionality.

### PR DESCRIPTION
**Pull request by codethulu**
Applicable spec: N/A

### Overview

Include `gh` and `unzip` functionality from the Ubuntu Archives. Included some simple checks in the e2e test file, to ensure the packages have been installed correctly.

### Rationale

This functionality was not currently present, and has been requested as an addition.

### Juju Events Changes

Juju events should not be affected by this change.

### Module Changes

No changes to any module functionality.

### Library Changes

Charm libraries are unchanged.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [ ] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

This PR does not warrant any new charm documentation due to the nature of this change.

### CODE REVIEW

**Review by arturo-seijas — COMMENTED** [review]
Can you add a title and PR description?

**Comment by cbartz** [comment]
> Some checks were unsuccessful - apparently this sometimes occurs. They do not seem related to any changes made. I am quite confident that it installs the packages correctly.

@codethulu  I ran them again - now they passed. The integration tests are unfortunately quite unstable, so unrelated failures may occur.

**Comment by codethulu** [comment]
Hi @cbartz I have updated the title and description, let me know if that looks alright now.

**Review by arturo-seijas — APPROVED** [review]

**Comment by codethulu** [comment]
Made requested changes 👍

**Comment by cbartz** [comment]
@jdkandersson The gh version installed here is 2.4.0+dfsg1 (2022-03-23 Ubuntu 2.4.0+dfsg1-2) compared to 2.43.1
 on the gh hosted ones. This could lead to users complaining if the client behaves differently.

**Comment by jdkandersson** [comment]
> @jdkandersson The gh version installed here is 2.4.0+dfsg1 (2022-03-23 Ubuntu 2.4.0+dfsg1-2) compared to 2.43.1 on the gh hosted ones. This could lead to users complaining if the client behaves differently.

How hard would it be to update that package to the latest version please?

File: scripts/build-image.sh [hunk 1]
```diff
@@ -96,7 +96,7 @@ retry '/snap/bin/lxc exec builder -- /usr/bin/nslookup github.com' 'Wait for net
 
 /snap/bin/lxc exec builder -- /usr/bin/apt-get update
 /snap/bin/lxc exec builder --env DEBIAN_FRONTEND=noninteractive -- /usr/bin/apt-get upgrade -yq
-/snap/bin/lxc exec builder --env DEBIAN_FRONTEND=noninteractive -- /usr/bin/apt-get install docker.io npm python3-pip shellcheck jq wget -yq
+/snap/bin/lxc exec builder --env DEBIAN_FRONTEND=noninteractive -- /usr/bin/apt-get install docker.io npm python3-pip shellcheck jq wget unzip gh -yq
```
Comment (jdkandersson) [inline scripts/build-image.sh:99]: As noted in a comment, I think `gh` might not be well maintained in the apt package, could we take that over and make sure it is the latest please?

Comment (mthaddon) [reply scripts/build-image.sh:99]: Not quite sure what you mean by taking it over. This is a package in the ubuntu archives, and versions of packages in the archives aren't updated outside of security fixes (see https://wiki.ubuntu.com/StableReleaseUpdates). We could create a PPA that has a more up to date version, but we'd be responsible for maintaining that.

Comment (mthaddon) [reply scripts/build-image.sh:99]: Let's go ahead with this as is for now, and open a separate task to make this more up to date (we'll need to decide whether to create a PPA for this or a snap).

**Review by mthaddon — APPROVED** [review]

**Comment by mthaddon** [comment]
Closing in favour of https://github.com/canonical/github-runner-operator/pull/222

**Pull request closed without merging**